### PR TITLE
Update application.yml with suggestions from TOC issue #1461

### PR DESCRIPTION
**Pull request by angellk**
Updated per @edrob999's suggestions and to address the suggestions brought up by @TheFoxAtWork 

Outstanding is reviewing and further clarifying the language to align with the GTR/DTR questions.

xref: https://github.com/cncf/toc/issues/1461

### CODE REVIEW

**Review by kevin-wangzefeng — APPROVED** [review]
LGTM, thanks

**Review by TheFoxAtWork — APPROVED** [review]
Lgtm thank you!!!

**Review by dims — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/application.yml [hunk 1]
```diff
@@ -173,14 +173,14 @@ body:
...
+    label: Project Domain Technical Review
+    description: Has your project engaged with the domain specific TAG(s) to increase awareness through a presentation and/or completing the Day 0 portion of the [General Technical Review](https://github.com/cncf/toc/blob/main/tags/resources/toc-supporting-guides/general-technical-questions.md) questionnaire? If so, please link meeting notes, recordings and/or completed [General Technical Review](https://github.com/cncf/toc/blob/main/tags/resources/toc-supporting-guides/general-technical-questions.md) questionnaire as applicable.
   validations:
     required: false
 - type: textarea
   attributes:
-    label: Project champions
-    description: Please list any people who are part of CNCF leadership (TOC, TAGS, etc.) who can endorse or answer questions about your project.
+    label: CNCF Contacts
+    description: Please list any people who are part of CNCF leadership (TOC, TAGS, etc.) who are familiar with, and can questions about your project.
```
Comment (TheFoxAtWork) [inline .github/ISSUE_TEMPLATE/application.yml:183]:
```suggestion
    description: Please list any people who are part of CNCF leadership (TOC, TAGS, etc.) who are familiar with, and can answer questions about your project.
```

**Review by TheFoxAtWork — CHANGES_REQUESTED** [review]
Incorporating "answer"

**Review by angellk — COMMENTED** [review]
lgtm

**Review by johnaAr555 — APPROVED** [review]

**Pull request merged**